src/tests/field/fctest_field_wrap_gpu.F90

```diff
@@ -90,39 +90,45 @@ module fcta_Field_wrap_device_fixture
   real(c_double), allocatable :: existing_data(:,:,:,:)
   real(c_double), pointer :: fview(:,:,:)
   type(atlas_Field) :: field
-  integer(c_int) :: i,j,k,l
-  write(0,*) "test_field_wrapdataslice [skipped]" ! NOT DONE YET !!!
-  return ! SKIP THIS TEST !!!
-  allocate( existing_data(4,3,2,5) )
+  integer(c_int) :: i, j, k, l, k_idx
+  integer(c_int), parameter :: Ni = 8, Nj = 2, Nk = 3, Nl = 4
+  write(0,*) "test_field_wrapdataslice"
+  allocate( existing_data(Ni, Nj, Nk, Nl) )
 
   existing_data = -1.
 
-  field = atlas_Field(existing_data(:,:,1,:))
-  call field%data(fview)
+  do l = 1, Nl
+    do k = 1, Nk
+      do j = 1, Nj
+        do i = 1, Ni
+          existing_data(i,j,k,l) = real(1000 * i+100 * j + 10*k + l, c_double)
+        enddo
+      enddo
+    enddo
+  enddo
 
-  !call field%allocate_device()
-  !call field%update_device()
+  k_idx = 1
+  field = atlas_Field(existing_data(:,:,k_idx,:))
+  call field%allocate_device()
+  call field%update_device()
+  call field%device_data(fview)
 
-  !!$acc data present(fview)
-  !!$acc parallel loop
-  do i=1,4
-    do j=1,3
-      do k=1,2
-        do l=1,5
-          fview(i,j,l) = 1000*i+100*j+10*1+l
-        enddo
+  !$acc parallel loop deviceptr(fview)
```

Comment:
I'm afraid that this `deviceptr(fview)` with fview having the `POINTER` attribute  is non-standard OpenACC and will not compile with Cray compiler. We need to create a separate subroutine just for kernels with deviceptr  😞, where the deviceptr is passed as argument without `POINTER` attribute.

Reply:
This was missing. Thank you for pointing it out. I have added kernel subroutines to call ACC pragmas. See 5dc24afbb.